src/main/java/org/codehaus/mojo/versions/DependencyUpdatesReport.java

```diff
@@ -184,42 +219,17 @@ else if ( "xml".equals( format ) )
     }
 
     /**
-     * Returns a set of dependencies where the dependencies which are defined in the dependency management section have
-     * been filtered out.
+     * Compares two dependencies with each other
      *
-     * @param dependencies The set of dependencies.
-     * @param dependencyManagement The set of dependencies from the dependency management section.
-     * @return A new set of dependencies which are from the set of dependencies but not from the set of dependency
-     *         management dependencies.
-     * @since 1.0-beta-1
+     * @return true if the two dependcies match
      */
-    private static Set<Dependency> removeDependencyManagment( Set<Dependency> dependencies, Set<Dependency> dependencyManagement )
+    private boolean match( Dependency dep, Dependency dmDep )
     {
-        Set<Dependency> result = new TreeSet<>( new DependencyComparator() );
-        for ( Dependency c : dependencies )
-        {
-            boolean matched = false;
-            Iterator<Dependency> j = dependencyManagement.iterator();
-            while ( !matched && j.hasNext() )
-            {
-                Dependency t = j.next();
-                if ( StringUtils.equals( t.getGroupId(), c.getGroupId() )
-                    && StringUtils.equals( t.getArtifactId(), c.getArtifactId() )
-                    && ( t.getScope() == null || StringUtils.equals( t.getScope(), c.getScope() ) )
-                    && ( t.getClassifier() == null || StringUtils.equals( t.getClassifier(), c.getClassifier() ) )
-                    && ( c.getVersion() == null || t.getVersion() == null
-                        || StringUtils.equals( t.getVersion(), c.getVersion() ) ) )
-                {
-                    matched = true;
-                    break;
-                }
-            }
-            if ( !matched )
-            {
-                result.add( c );
-            }
-        }
-        return result;
+        return StringUtils.equals( dmDep.getGroupId(), dep.getGroupId() ) && StringUtils.equals( dmDep.getArtifactId(),
+                dep.getArtifactId() ) && ( dmDep.getScope() == null || StringUtils.equals( dmDep.getScope(),
+                dep.getScope() ) ) && ( dmDep.getClassifier() == null || StringUtils.equals( dmDep.getClassifier(),
+                dep.getClassifier() ) ) && ( dep.getVersion() == null || dmDep.getVersion() == null
+                || StringUtils.equals( dmDep.getVersion(), dep.getVersion() ) );
     }
```

Comment:
- why not `Object.equals` from jdk
- try  split line by `&&` ...

Reply:
> why not Object.equals

Okay, done.